app/experimenter/nimbus-ui/src/components/PageEditOverview/index.tsx

```diff
@@ -3,35 +3,74 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import React, { useCallback, useState } from "react";
-import { RouteComponentProps, useParams } from "@reach/router";
+import { navigate, RouteComponentProps, useParams } from "@reach/router";
 import AppLayoutWithSidebar from "../AppLayoutWithSidebar";
 import { useExperiment } from "../../hooks";
 import HeaderEditExperiment from "../HeaderEditExperiment";
 import PageExperimentNotFound from "../PageExperimentNotFound";
 import PageLoading from "../PageLoading";
 import FormOverview from "../FormOverview";
+import { useMutation } from "@apollo/client";
+import { UpdateExperimentInput } from "../../types/globalTypes";
+import { updateExperimentOverview_updateExperimentOverview as UpdateExperimentOverviewResult } from "../../types/updateExperimentOverview";
+import { UPDATE_EXPERIMENT_OVERVIEW_MUTATION } from "../../gql/experiments";
+import { SUBMIT_ERROR } from "../../lib/constants";
 
 type PageEditOverviewProps = {} & RouteComponentProps;
 
 const PageEditOverview: React.FunctionComponent<PageEditOverviewProps> = () => {
-  // TODO: EXP-462 Get this from constants / config loaded at app start?
-  const applications = ["firefox-desktop", "fenix", "reference-browser"];
+  const [updateExperimentOverview, { loading: mutationLoading }] = useMutation<
+    { updateExperimentOverview: UpdateExperimentOverviewResult },
+    { input: UpdateExperimentInput }
+  >(UPDATE_EXPERIMENT_OVERVIEW_MUTATION);
 
   const { slug } = useParams();
-  const { experiment, notFound, loading } = useExperiment(slug);
-  const [submitErrors /* setSubmitErrors */] = useState<Record<string, any>>(
-    {},
+  const { experiment, notFound, loading: experimentLoading } = useExperiment(
+    slug,
   );
+  const [submitErrors, setSubmitErrors] = useState<Record<string, any>>({});
 
-  const onFormSubmit = useCallback(() => {
-    console.log("SUBMIT TBD");
-  }, []);
+  const onFormSubmit = useCallback(
+    async (
+      { name, hypothesis, application, publicDescription }: Record<string, any>,
+      resetForm: Function,
+    ) => {
+      try {
+        const result = await updateExperimentOverview({
+          variables: {
+            input: {
+              id: experiment.id,
+              name,
+              hypothesis,
+              application,
+              publicDescription,
+            },
+          },
+        });
+
+        if (!result.data?.updateExperimentOverview) {
+          throw new Error("Save failed, no error available");
+        }
+
+        const { message } = result.data.updateExperimentOverview;
+
+        if (message !== "success" && typeof message === "object") {
+          return void setSubmitErrors(message);
+        }
+
+        resetForm();
```

Comment:
I think we want to remove L61 or rework it somehow. When I make an edit and click "Save," the old value is repopulated in the field. If we just remove it though, the "Save" button doesn't disable with "Next" re-enabled.

Reply:
Fixed